From 45f8b159261de20f16e69701f16fed7f29100480 Mon Sep 17 00:00:00 2001
From: Martin Rotter <rotter.martinos@gmail.com>
Date: Tue, 24 Jun 2025 07:21:49 +0200
Subject: [PATCH] fix long-standing bug #1654

---
 .../gui/webviewers/qtextbrowser/textbrowserviewer.cpp      | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/src/librssguard/gui/webviewers/qtextbrowser/textbrowserviewer.cpp b/src/librssguard/gui/webviewers/qtextbrowser/textbrowserviewer.cpp
index dd21178a8..59774dd48 100644
--- a/src/librssguard/gui/webviewers/qtextbrowser/textbrowserviewer.cpp
+++ b/src/librssguard/gui/webviewers/qtextbrowser/textbrowserviewer.cpp
@@ -56,10 +56,9 @@ TextBrowserViewer::TextBrowserViewer(QWidget* parent)
 }
 
 TextBrowserViewer::~TextBrowserViewer() {
-  if (m_resourceDownloaderThread->isRunning()) {
-    m_resourceDownloaderThread->quit();
-  }
-
+  m_resourceDownloaderThread->quit();
+  m_resourceDownloaderThread->wait();
+  m_resourceDownloaderThread->deleteLater();
   m_resourceDownloader->deleteLater();
 }
 
