diff --git a/src/librssguard/miscellaneous/application.cpp b/src/librssguard/miscellaneous/application.cpp
index c4ecc718e..c8f732f55 100644
--- a/src/librssguard/miscellaneous/application.cpp
+++ b/src/librssguard/miscellaneous/application.cpp
@@ -910,7 +910,6 @@ void Application::onAboutToQuit() {
   // Make sure that we obtain close lock BEFORE even trying to quit the application.
   const bool locked_safely = feedUpdateLock()->tryLock(4 * CLOSE_LOCK_TIMEOUT);
 
-  QCoreApplication::processEvents();
   qDebugNN << LOGSEC_CORE << "Cleaning up resources and saving application state.";
 
   if (locked_safely) {
