diff --git a/src/librssguard/network-web/adblock/adblockmanager.h b/src/librssguard/network-web/adblock/adblockmanager.h
index 3b1214beb..449b49cd1 100644
--- a/src/librssguard/network-web/adblock/adblockmanager.h
+++ b/src/librssguard/network-web/adblock/adblockmanager.h
@@ -10,7 +10,7 @@
 #include <QProcess>
 
 #define CLIQZ_ADBLOCKED_PACKAGE "@ghostery/adblocker"
-#define CLIQZ_ADBLOCKED_VERSION "2.5.1"
+#define CLIQZ_ADBLOCKED_VERSION "2.12.5"
 
 class QUrl;
 class AdblockRequestInfo;
